climt/_core/constants.py

```diff
@@ -0,0 +1,303 @@
+                               'heat_capacity_of_vapor_phase',
+                               'specific_enthalpy_of_vapor_phase',
+                               'gas_constant_of_vapor_phase',
+                               'latent_heat_of_condensation',
+                               'latent_heat_of_fusion',
+                               'density_of_solid_phase_as_ice',
+                               'density_of_solid_phase_as_snow',
+                               'heat_capacity_of_solid_phase_as_ice',
+                               'heat_capacity_of_solid_phase_as_snow',
+                               'thermal_conductivity_of_solid_phase_as_ice',
+                               'thermal_conductivity_of_solid_phase_as_snow',
+                               'thermal_conductivity_of_liquid_phase',
+                               'freezing_temperature_of_liquid_phase',)),
+
+    atmospheric_constants=set(('heat_capacity_of_dry_air_at_constant_pressure',
+                               'gas_constant_of_dry_air',
+                               'thermal_conductivity_of_dry_air',
+                               'reference_air_pressure')),
+
+    stellar_constants=set(('stellar_irradiance',)),
+
+    oceanographic_constants=set(()),
+
+    chemical_constants=set(('heat_capacity_of_water_vapor_at_constant_pressure',
+                            'density_of_liquid_water',
+                            'gas_constant_of_water_vapor',
+                            'latent_heat_of_vaporization_of_water',
+                            'heat_capacity_of_liquid_water',
+                            'latent_heat_of_fusion_of_water',)),
+
+    miscellaneous_constants=set(())
+)
+
+
+class ConstantLibrary(object):
+    """
+    Maintain constant values and categories. Currently available
+    constants are accessible as :code:`constant_library.current_constants`.
+    """
+
+    def __init__(self):
+
+        self.init_constant_library()
+
+    def init_constant_library(self):
+        """
+        Initialise the constant library from sympl's :code:`default_constants`.
+        """
+
+        self.current_constants = deepcopy(default_constants)
+
+        self.constant_categories = deepcopy(default_categories)
+
+        for climt_name in constant_name_map.keys():
+            self.current_constants[climt_name] = deepcopy(
+                default_constants[constant_name_map[climt_name]])
+
+    def reset_constant_library(self):
+        """
+        Remove all additions to constant library that have happened
+        during runtime.
+        """
+
+        self.init_constant_library()
+
+    def _list_constants(self, constant_list):
+
+        for quantity in constant_list:
+            print("{}: {} {}".format(quantity,
+                                     self.current_constants[quantity].values,
+                                     self.current_constants[quantity].units))
+            print()
+
+    def list_available_constants(self):
```

Comment:
I was going to ask why it's necessary to have this object at all, but this method alone is probably good justification. Very useful!